tools/differ/helper.py

```diff
@@ -0,0 +1,88 @@
+import glob
+import os
+import pandas as pd
+import re
+
+from google.cloud.storage import Client
+
+
+def load_mcf_file(file: str) -> pd.DataFrame:
+    """ Reads an MCF text file and returns it as a dataframe."""
+    mcf_file = open(file, 'r', encoding='utf-8')
+    mcf_contents = mcf_file.read()
+    mcf_file.close()
+    # nodes separated by a blank line
+    mcf_nodes_text = mcf_contents.split('\n\n')
+    # lines seprated as property: constraint
+    mcf_line = re.compile(r'^(\w+): (.*)$')
+    mcf_nodes = []
+    for node in mcf_nodes_text:
+        current_mcf_node = {}
+        for line in node.split('\n'):
+            parsed_line = mcf_line.match(line)
+            if parsed_line is not None:
+                current_mcf_node[parsed_line.group(1)] = parsed_line.group(2)
+        if current_mcf_node and current_mcf_node[
+                'typeOf'] == 'dcid:StatVarObservation':
+            mcf_nodes.append(current_mcf_node)
+    df = pd.DataFrame(mcf_nodes)
+    return df
+
+
+def load_mcf_files(path: str) -> pd.DataFrame:
+    """ Loads all sharded mcf files in the given directory and 
+    returns a single combined dataframe."""
+    df = pd.DataFrame()
+    filenames = glob.glob(path + '.mcf')
+    for filename in filenames:
+        df2 = load_mcf_file(filename)
+        # Merge data frames, expects same headers
+        df = pd.concat([df, df2])
```

Comment:
collecting all dfs into a list and a single concat at the end may save some copies.

Reply:
Done